lib/msf/core/exploit/remote/http/spip.rb

```diff
@@ -0,0 +1,41 @@
+# -*- coding: binary -*-
+
+module Msf
+module Exploit::Remote::HTTP::Spip
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to Spip install', '/'])
+    ])
+  end
+
+  # Determine Spip version
+  #
+  # @return [Rex::Version] Version as Rex::Version
+  def spip_version
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri'    => normalize_uri(target_uri.path, "spip.php")
+    )
+
+    return unless res
+    
+    version = res.get_html_document.at('head/meta[@name="generator"]/@content')&.text
+
+    if version.nil? && res.headers['Composed-By'] =~ /SPIP (.*) @/
+      version = ::Regexp.last_match(1)
+    end
```

Comment:
I see you're pulling the version from two different places here. Is there any context you can easily add in a comment about when one technique should work versus the other? What I'm wondering is if there's a version where prior one technique is used and after another is used or if it's dependent on some kind of configuration setting.

Reply:
I think it depends on the configuration/version of Spip, but I don't know the details. Maybe @laluka knows?

Reply:
Henlo !
It's all depends on the config, whether files are accessible, headerd, or reflected html content, there's sadly no thumb rule :/

Nice addon btw, and have a nice day! 🌹